File: 02-java-spring-boot/src/main/java/com/tqi/checkout/Main.java
//@file:JvmName("Application")
package com.tqi.checkout;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@RestController
public class Main {

    public static void main(String[] args) {
        // COMPARAÇÃO COM KOTLIN: No Java, o ponto de entrada da aplicação obrigatoriamente 
        // precisa estar envelopado dentro de um método 'public static void main' contido em uma classe.
        SpringApplication.run(Main.class, args);
    }
    @GetMapping("/status")
    public String status() {
        return "Servidor online!";
    }
}

File: 02-java-spring-boot/src/main/kotlin/com/tqi/checkout/Main.kt
package com.tqi.checkout
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Bean
import com.tqi.checkout.service.CheckoutService
import com.tqi.checkout.domain.strategy.PixPagamento

@SpringBootApplication
class Main {
    @Bean fun run(s: CheckoutService, p: PixPagamento) = CommandLineRunner { s.finalizarPedido(400.0, p) }
}
fun main(args: Array<String>) { runApplication<Main>(*args) }
